docs/02-architecture/06-client-side-architecture.md

```diff
@@ -0,0 +1,285 @@
+- Commercial Javascript
+- Enhanced Javascript
+
+And then specific Javascript bootstraps used on different pages and required into the app when needed - Article, Article Minute, Crosswords, Liveblog, Gallery, Trail, Profile, Sudoku, Image content, Facia, Football, Preferences, Membership, Ophan, Admin, Main Media, Video Embed, Accessibility.
+
+### General structure
+
+See below for quick descriptions.
+
+- [- main.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/main.scala.html)
+	- [head.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/head.scala.html)
+		- [inlineJsBlocking.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/inlineJSBlocking.scala.html) - Scripts required to render the page correctly on first paint.
+			- Polyfills for json2, es5-html5, RAF, classlist
+			- Curl config
+			- shouldEnhance
+			- pageConfig
+			- applyRenderConditions
+			- loadFonts
+			- enable non blocking stylesheets
+			- **Load the main app async**
+				- app.js
+				- commercial.js
+				- enhanced/main.js
+			- Cloudwatch beacon
+		- [inlineJSNonBlocking.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/inlineJSNonBlocking.scala.html)
+			- getUserData.js
+			- detectAdblock 
+			- showUserName
+			- ophanConfig
+		
+### Inline blocking JS 
+
+The inline blocking JS is in the head of the document and will block render until it has finished executing. Each of these scripts are required to render the page correctly on first paint.
+		
+#### Curl config
+
+Contains the [config](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/blocking/curlConfig.scala.js) for the [curl AMD module loader](https://github.com/cujojs/curl). You can find the aliases for certain JS modules in here such as `fastdom` which helps you avoid having to add the full path for certain modules.
+
+#### shouldEnhance
+
+The [shouldEnhance JS](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/blocking/shouldEnhance.scala.js) defines whether we should run the enhanced Javascript. 
+
+There are ways to force enhancement, so take a look at the JS, but something of note is that we don't enhance devices running iOS < 8 on all pages or on all iPads on Fronts.
+
+#### pageConfig
+
+If you put `guardian.config` in your console, you will see the JS config containing information about analytics, modules, switches, tests and the page. This config object is initially populated serverside and topped-up client side (i.e.: information about the user is only available client-side)
+
+The initial config structure is defined in [config.scala.js](https://github.com/guardian/frontend/blob/master/common/app/templates/inlineJS/blocking/config.scala.js) and [javaScriptConfig.scala.js](https://github.com/guardian/frontend/blob/master/common/app/templates/javaScriptConfig.scala.js). 
+
```

Comment:
Is it worth mentioning here that `config.scala.js` also sets the `isModernBrowser` flag, which determines whether a user's browser cuts the mustard? Assuming this is actually used anywhere 😄